consumer.go

```diff
@@ -785,11 +785,20 @@ func (c *Consumer) isPotentialExtraTopic(topic string) bool {
 	return false
 }
 
-func (c *Consumer) refreshMetadata() error {
-	err := c.client.RefreshMetadata()
-	if err == sarama.ErrTopicAuthorizationFailed {
-		// maybe we didn't have authorization to describe all topics
+func (c *Consumer) refreshMetadata() (err error) {
+	if c.client.config.Metadata.Full {
```

Comment:
there can be a case where c.client is nil and it might panic , it is better to check just to make sure c.client is not nil .